lib/helpers/response-handler.js

```diff
@@ -6,13 +6,22 @@
  * @return {Promise} resolves to the wrapped read response
  */
 function wrapResponse(response, reader) {
-  return response[reader]()
-  .then(data => ({
+  const res = {
     headers   : response.headers,
     status    : response.status,
-    statusText: response.statusText,
-    data
-  }));
+    statusText: response.statusText
+  };
+
+  if (reader === 'stream') {
+    res.data = response.body.getReader();
```

Comment:
Yeah the `raw` `bodyType` I like, is going to go through the middlewares and it will work unless they try to modify the `data`, but users can still work with headers etc.